src/pages/booklist/hooks/useTextbooks.tsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useMemo, useState } from 'react';
+
+import { useParams } from 'react-router';
+
+import { useSavedCourses } from 'lib/hooks/useSavedCourses';
+
+import { getCourseTextbooks, TextbookInfo } from '../api/getCourseTextbooks';
+
+type TextbookResult = { status: 'loaded'; textbookInfo: TextbookInfo[] } | { status: 'loading' };
+
+export const useTextbooks = (): TextbookResult => {
+  const [result, setResult] = useState<TextbookResult>({ status: 'loading' });
+  const { courses } = useSavedCourses();
+  const { term } = useParams();
```

Comment:
I wouldn't pass in term like this as we get a hard dependency on using this hook